quickwit/quickwit-search/src/collector.rs

```diff
@@ -293,25 +233,64 @@ enum AggregationSegmentCollectors {
 pub struct QuickwitSegmentCollector {
     num_hits: u64,
     split_id: String,
-    score_extractor: SortingFieldExtractorPair,
+    score_extractors: Vec<SortingFieldExtractorComponent>,
```

Comment:
See comment below this is not a Vec today for good reasons.
Did you need it to be a Vec?

Reply:
having a pair of options everywhere makes most code weird, plus limit us in term of feature. I'd prefer to have something that can be iterated over, and ideally which can grow beside two elements. Would using a smallvec, to inline up to N elements (with N somewhere somewhere in 3..=5) be fine? It wouldn't allocate for most queries, and would also check my boxes

Reply:
See comment below. We cannot do this with without benchmarking it, and let's not do this in this PR.

Reply:
ftr, when collecting half a million docs, here is how long it took (not great for vec and smallvec)
```
double option:
test collector::tests::bench_collection ... bench:  95,939,300 ns/iter (+/- 2,402,897)
vec:
test collector::tests::bench_collection ... bench: 131,689,753 ns/iter (+/- 2,208,591)
smallvec:
test collector::tests::bench_collection ... bench: 137,068,856 ns/iter (+/- 1,300,872)
```